otel-extension/src/main/scala/io/scalac/mesmer/otelextension/instrumentations/zio/ZIOMetrics.scala

```diff
@@ -25,16 +26,35 @@ object ZIOMetrics {
       .gaugeBuilder(metricName(metricKey.name))
       .buildWithCallback(_.record(unsafeGetGaugeValue(metricKey), buildAttributes(metricKey.tags)))
 
+  def registerHistogramSyncMetric(metricKey: MetricKey.Histogram): DoubleHistogram = {
+    val underlying = meter
+      .histogramBuilder(metricName(metricKey.name))
+      .build()
+    // workaround for classloading issues
+    val fn1 = (double: Double) => underlying.record(double)
+    val fn2 = (double: Double, tags: Set[MetricLabel]) => underlying.record(double, buildAttributes(tags))
+    new DoubleHistogram {
+      def record(double: Double): Unit                         = fn1(double)
+      def record(double: Double, tags: Set[MetricLabel]): Unit = fn2(double, tags)
+    }
+  }
+
   private def buildAttributes(metricLabels: Set[MetricLabel]): Attributes = {
     val builder = Attributes.builder()
     metricLabels.foreach { case MetricLabel(key, value) => builder.put(AttributeKey.stringKey(key), value) }
     builder.build()
   }
 
   private def unsafeGetCounterValue(metricKey: MetricKey[MetricKeyType.Counter]): Double =
-    ConcurrentMetricRegistryClient.get(metricKey).get().count
+    client.get(metricKey).get().count
 
   private def unsafeGetGaugeValue(metricKey: MetricKey[MetricKeyType.Gauge]): Double =
-    ConcurrentMetricRegistryClient.get(metricKey).get().value
+    client.get(metricKey).get().value
+}
 
+object ZIOMetrics {
+  trait DoubleHistogram {
```

Comment:
Unless I'm missing something, if you make `buildAttributes` implicit, there's no need for the custom trait and you can use the io.opentelemetry.metrics.DoubleHistogram object. Did you consider that? 

On the other hand implicit may be ugly/hard(er) to migrate to java in the future(?)

Reply:
The need for this trait is explained in this comment - https://github.com/ScalaConsultants/mesmer/pull/650#discussion_r1150430960.